spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/MongoRepository.java

```diff
@@ -18,43 +18,50 @@
 import java.io.Serializable;
 import java.util.List;
 
+import org.springframework.data.domain.Example;
 import org.springframework.data.domain.Sort;
 import org.springframework.data.repository.NoRepositoryBean;
 import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.QueryByExampleExecutor;
 
 /**
  * Mongo specific {@link org.springframework.data.repository.Repository} interface.
- * 
+ *
  * @author Oliver Gierke
  * @author Christoph Strobl
  * @author Thomas Darimont
+ * @author Mark Paluch
  */
 @NoRepositoryBean
-public interface MongoRepository<T, ID extends Serializable> extends PagingAndSortingRepository<T, ID> {
```

Comment:
Some formatting related changes. Adding the `@Override`s is not a bad idea, but we should keep them in a separate commit or simply postpone them.